src/lib/rest/rest.cpp

```diff
@@ -1220,6 +1220,18 @@ static int connectionTreat
     ciP->httpHeader.push_back("Fiware-Correlator");
     ciP->httpHeaderValue.push_back(ciP->httpHeaders.correlator);
 
+    if ((ciP->httpHeaders.contentLength > PAYLOAD_MAX_SIZE) && (ciP->apiVersion == V2))
```

Comment:
NGSI v1 requests are tricky.
Each type of request has a different response payload.
Not sure it is worth the effort to implement this for NGSI v1
Out-of-NGSI requests have very small if any payload at all.
So, not very interesting, could be implemented though. Easily.